### PR TITLE
Egg + Microwave = egg splat.

### PR DESCRIPTION
**Pull request by arch-linux**
## About the PR

I have created three additional sprites consistent with an egg exploding in a microwave. Before this, the egg would create a meat explosion showing the typical blood-gore microwave sprite.

Egg Before - Ew right?
egg before

Egg After - Seems right!
egg after

Please see the proposed egg sprites in action. 
https://youtu.be/WeTesUsEAcY

## Why's this needed?

This is going to be my first installment of slightly more realistic/hilarious kitchen functions. 
Plus? have you ever seen an egg in a microwave? **dont try this at home**

https://youtu.be/PfyVRhL2Dgw?t=131

## Changelog

archlinuxusa: 
* Eggs no longer produce a meaty explosion in the microwave, splattered egg will now splatter all over your microwave... Yummy.

### CODE REVIEW

File: code/obj/machinery/microwave.dm [hunk 1]
```diff
@@ -281,6 +281,30 @@ Please clean it before use!</TT><BR>
...
+						src.icon_state = "mw1"
+						src.updateUsrDialog()
+						src.egg_amount = 0 //Clear all the values as this crap is what makes the mess inside!!
+						src.flour_amount = 0
+						src.water_amount = 0
+						src.humanmeat_amount = 0
+						src.monkeymeat_amount = 0
+						src.synthmeat_amount = 0
+						src.donkpocket_amount = 0
+						sleep(40) // Half way through
```
Comment (ZeWaka) [inline code/obj/machinery/microwave.dm:296]: Might be good to do some nullchecking after these sleeps.

File: code/obj/machinery/microwave.dm [hunk 2]
```diff
@@ -257,6 +257,29 @@ Please clean it before use!</TT><BR>
 							src.extra_item = null // And the extra item
 							cooked_item = R.creates // Store the item that will be created
 
+						else if(src.egg_amount > 0) // egg was inserted alone
```
[the remark this answers is not in the document]
Comment (arch-linux) [reply code/obj/machinery/microwave.dm:260]: Your absolutely right, that was dumb.

**Comment by ZeWaka** [comment]
Merge conflicts

**Comment by arch-linux** [comment]
After extensive battling with learning how to git merge on my fork properly i believe this build is ready. **The only two files that should be merged into master are the kitchen.dmi and the microwave.dm.** 
I apologize for all of the commits; my VSstudio suffered from a bug after attempting to rebase from master. I will be more careful in the future.

**Comment by arch-linux** [comment]
- two hours of battling with git (learned an ass-day amount)
- ready to be merged.

**Pull request merged**